File: 2025/day_01/01.py

if __name__ == "__main__":

    filename = "01-input.txt"

    with open(filename) as f:
        lines = f.read().splitlines()

    dial = 50
    delta = 0
    zero_count = 0

    for l in lines:
        if l[0] == "L":
            delta = -1
        else:
            delta = 1
        count = int(l[1:])
        for i in range(count):
            dial += delta 
            if dial == -1:
                dial = 99 
            elif dial == 100:
                dial = 0
            if dial == 0:
                zero_count += 1
    
    print("zero count:", zero_count)

File: 2025/day_01/01.rs
use std::fs;

fn main() {
    let contents = fs::read_to_string("01-input.txt").expect("Cannot read file");
    let entries: Vec<&str> = contents.trim().split('\n').collect();

    let mut dial: i32 = 50;
    let mut zero_count: i32 = 0;

    for entry in entries {
        let (direction, amount) = entry.split_at(1);
        let rotations: i32 = amount.trim().parse().expect("Not a valid number.");
        let increment: i32;
        if direction == "L" {
            increment = -1;
        }
        else {
            increment = 1;
        }
        for _i in 0..rotations {
            dial += increment;
            if dial == -1 {
                dial = 99;
            }
            else if dial == 100 {
                dial = 0;
            }
            if dial == 0 {
                zero_count += 1;
            }
        }
    }
    println!("{zero_count}");
}